server/pkg/item/item.go

```diff
@@ -69,6 +70,22 @@ func (i *Item) Thread() ThreadID {
 	return i.thread
 }
 
+func (i *Item) UpdateField(fid *id.FieldID, v *value.Multiple) {
+	if fid == nil || v == nil {
+		return
+	}
+	_, ii, ok := lo.FindIndexOf(i.fields, func(f *Field) bool {
+		return f.FieldID() == *fid
+	})
+	if !ok {
+		return
+	}
+	if v.Type() != i.Field(*fid).Type() {
+		return
+	}
+	i.fields[ii].value = v
+}
```

Comment:
This method is not needed I think

```go
i.UpdateFields([]*Field{item.NewField(fid, v)})
```

Basically, the private fields of a struct should only be accessible by that struct's methods. Accessing it from any other way will make the code very difficult to understand and will create bugs.

Also note that item.Field is an immtuable struct because there is no setter. However, this implementation breaks the immutability of the field.

Reply:
Understood, Thanks for the comment!
Done!

Reply:
Removed, Thanks for the comment!
Done!